File: Medium/2026-01-29-2976-Minimum-Cost-to-Convert-String-I/solution.java
import java.util.HashMap;
import java.util.Map;

public class Solution {
    public int minimumCost(String source, String target, String[] original, String[] changed, int[] cost) {
        int n = source.length();
        Map<String, Map<String, Integer>> graph = new HashMap<>();
        
        for (int i = 0; i < original.length; i++) {
            graph.putIfAbsent(original[i], new HashMap<>());
            graph.get(original[i]).put(changed[i], Math.min(graph.get(original[i]).getOrDefault(changed[i], Integer.MAX_VALUE), cost[i]));
        }
        
        int totalCost = 0;
        for (int i = 0; i < n; i++) {
            char srcChar = source.charAt(i);
            char tgtChar = target.charAt(i);
            if (srcChar == tgtChar) continue;
            
            if (!graph.containsKey(String.valueOf(srcChar))) {
                return -1;
            }
            
            Map<String, Integer> changes = graph.get(String.valueOf(srcChar));
            if (!changes.containsKey(String.valueOf(tgtChar))) {
                return -1;
            }
            
            totalCost += changes.get(String.valueOf(tgtChar));
        }
        
        return totalCost;
    }
}

File: Medium/2026-01-29-2976-Minimum-Cost-to-Convert-String-I/solution.js
function minimumCost(source, target, original, changed, cost) {
    const n = source.length;
    const costMap = new Map();

    for (let i = 0; i < original.length; i++) {
        const key = original[i] + '->' + changed[i];
        if (!costMap.has(key)) {
            costMap.set(key, cost[i]);
        } else {
            costMap.set(key, Math.min(costMap.get(key), cost[i]));
        }
    }

    let totalCost = 0;

    for (let i = 0; i < n; i++) {
        if (source[i] === target[i]) continue;

        const key = source[i] + '->' + target[i];
        if (!costMap.has(key)) {
            return -1;
        }

        totalCost += costMap.get(key);
    }

    return totalCost;
}